### PR TITLE
[fix](Nereids) store user variable in connect context

### PR DESCRIPTION
**Pull request by keanji-x**
## Proposed changes

1. user variable should be case insensitive
3. user variable should be cleared after the connection reset

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by XieJiann** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/qe/ConnectContext.java [hunk 1]
```diff
@@ -301,6 +317,7 @@ public void init() {
         returnRows = 0;
         isKilled = false;
         sessionVariable = VariableMgr.newSessionVariable();
+        userVars = new ConcurrentHashMap<>();
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ConnectContext.java:320]: why concurrentHashMap?

File: fe/fe-core/src/main/java/org/apache/doris/qe/ConnectContext.java [hunk 2]
```diff
@@ -450,6 +467,64 @@ public void setEnv(Env env) {
         defaultCatalog = env.getInternalCatalog().getName();
     }
 
+    public void setUserVar(SetVar setVar) {
+        userVars.put(setVar.getVariable().toLowerCase(), setVar.getResult());
+    }
+
+    public @Nullable Literal getLiteralForUserVar(String varName) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ConnectContext.java:474]: varName always lower case?

Comment (XieJiann) [reply fe/fe-core/src/main/java/org/apache/doris/qe/ConnectContext.java:474]: user variable is case insensitive, so convert to lowercase

File: fe/fe-core/src/main/java/org/apache/doris/qe/ConnectContext.java (hunk 1 above)
Comment (XieJiann) [reply fe/fe-core/src/main/java/org/apache/doris/qe/ConnectContext.java:320]: this could be W/R concurrently

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Pull request merged**